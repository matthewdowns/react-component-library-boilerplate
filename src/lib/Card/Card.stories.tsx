import { Meta } from '@storybook/react';
import React from 'react';
import Card from './Card';
import CardProps from './Card.props';

export default {
    title: 'Card',
    component: Card,
    argTypes: {
        children: {
            table: {
                disable: true
            }
        },
        'data-testid': {
            table: {
                disable: true
            }
        }
    }
} as Meta<CardProps>;

const Template = ({ children, ...rest }: CardProps) => (
    <Card {...rest}>
        {children}
    </Card>
);

export const Default = Template.bind({});
Default.args = {
    children: (
        <>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in lacus eu massa efficitur dapibus ac et leo. Maecenas a nulla enim. Sed commodo eros ultrices, scelerisque augue tincidunt, maximus erat. Duis suscipit nulla at nulla vulputate ultricies. Fusce tempus porta dapibus. In laoreet eros nulla, non semper metus dignissim sit amet. Suspendisse ac tristique sem, vitae placerat nisl. Etiam ac egestas sem, venenatis dictum justo. In hac habitasse platea dictumst. Praesent justo nulla, scelerisque in accumsan eu, porta ac risus. Praesent sed leo nibh. Fusce finibus vulputate sapien, ac congue libero volutpat vitae.
        </>
    )
}

export const Title = Template.bind({});
Title.args = {
    children: (
        <>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in lacus eu massa efficitur dapibus ac et leo. Maecenas a nulla enim. Sed commodo eros ultrices, scelerisque augue tincidunt, maximus erat. Duis suscipit nulla at nulla vulputate ultricies. Fusce tempus porta dapibus. In laoreet eros nulla, non semper metus dignissim sit amet. Suspendisse ac tristique sem, vitae placerat nisl. Etiam ac egestas sem, venenatis dictum justo. In hac habitasse platea dictumst. Praesent justo nulla, scelerisque in accumsan eu, porta ac risus. Praesent sed leo nibh. Fusce finibus vulputate sapien, ac congue libero volutpat vitae.
        </>
    ),
    title: 'Title'
}
