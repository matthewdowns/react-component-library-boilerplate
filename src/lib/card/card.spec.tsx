import { render } from '@testing-library/react';
import React from 'react';
import Card from './card';
import CardProps from './card.props';

const children = <p>Children</p>;
const header = <p>Header</p>;
const footer = <p>Footer</p>;
const testid = 'card';

describe('Card', () => {
    it('matches snapshot', () => {
        const body = renderCard({ children, header, footer })
        expect(body.container).toMatchSnapshot()
    })

    it('renders props:children', () => {
        const body = renderCard({ children })
        const element = body.getByTestId(testid)
        expect(element.children.length).toBe(1)
        expect(element.children[0].textContent).toBe('Children')
    })

    it('renders props:header', () => {
        const body = renderCard({ header })
        const element = body.getByTestId(testid)
        expect(element.children.length).toBe(2)
        expect(element.children[0].textContent).toBe('Header')
    })

    it('renders props:footer', () => {
        const body = renderCard({ footer })
        const element = body.getByTestId(testid)
        expect(element.children.length).toBe(2)
        expect(element.children[1].textContent).toBe('Footer')
    })
})

const defaultProps: CardProps = {
    children,
    'data-testid': testid
}

function renderCard(props?: Partial<CardProps>) {
    return render(
        <Card
            {...defaultProps}
            {...props}
        />
    )
}
