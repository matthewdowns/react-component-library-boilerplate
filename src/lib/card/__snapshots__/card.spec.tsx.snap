// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  >
    <div
      class="card-header"
    >
      <p>
        Header
      </p>
    </div>
    <div
      class="card-body"
    >
      <p>
        Children
      </p>
    </div>
    <div
      class="card-footer"
    >
      <p>
        Footer
      </p>
    </div>
  </div>
</div>
`;
